lib/limiter/connlimiter.go

```diff
@@ -64,55 +62,62 @@ func (l *ConnectionsLimiter) WrapHandle(h http.Handler) {
 
 // AcquireConnection acquires connection and bumps counter
 func (l *ConnectionsLimiter) AcquireConnection(token string) error {
-	l.Lock()
-	defer l.Unlock()
-
 	if l.maxConnections == 0 {
 		return nil
 	}
 
+	l.Lock()
+	defer l.Unlock()
+
 	numberOfConnections, exists := l.connections[token]
 	if !exists {
 		l.connections[token] = 1
 		return nil
 	}
+
 	if numberOfConnections >= l.maxConnections {
-		return trace.LimitExceeded(
-			"too many connections from %v: %v, max is %v",
-			token, numberOfConnections, l.maxConnections)
+		return trace.LimitExceeded("too many connections from %v: %v, max is %v", token, numberOfConnections, l.maxConnections)
 	}
-	l.connections[token] = numberOfConnections + 1
+
+	l.connections[token]++
```

Comment:
This is a pessimization, technically? I hope the compiler is smart enough to see through it